pkg/core/managers/apis/mesh/dataplane_manager.go

```diff
@@ -0,0 +1,70 @@
+package mesh
+
+import (
+	"context"
+	core_mesh "github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	core_manager "github.com/Kong/kuma/pkg/core/resources/manager"
+	core_model "github.com/Kong/kuma/pkg/core/resources/model"
+	core_store "github.com/Kong/kuma/pkg/core/resources/store"
+	validators "github.com/Kong/kuma/pkg/core/validators/apis/mesh"
+	"github.com/pkg/errors"
+
+	"github.com/Kong/kuma/pkg/core/resources/model"
+	"github.com/Kong/kuma/pkg/core/resources/store"
+)
+
+type dataplaneManager struct {
```

Comment:
It's not strictly necessary to have a custom `dataplaneManager` right now (until we do validation that spans multiple resources).

Alternatively, we could expect every resource to implement `Validate() error` method. And one generic manager would be enough.

Reply:
changed to `Validate() error` on resource itself. Seems simpler for simple validation logic